src/io/invoca/Docker.groovy

```diff
@@ -0,0 +1,69 @@
+#!/usr/bin/groovy
+package io.invoca;
+
+/* args
+    build_args  list (optional)
+    dockerfile  string
+    image_name  string
+*/
+
+@NonCPS
+def buildCommand(Map args) {
+    def String build_args = ["", *args.build_args].join(' --build-arg ')
+    def String cmd
+    
+    // http://label-schema.org/rc1/
+    cmd = "docker build -t ${args.image_name}:${env.GIT_COMMIT} \
+            --label org.label-schema.schema-version=1.0 \
+            --label org.label-schema.build-date=`date -u +\"%Y-%m-%dT%H:%M:%SZ\"` \
+            --label org.label-schema.vcs-url=${env.GIT_URL}"
+    
+    if (build_args) {
+        cmd += " ${build_args}"
+    }
+
+    cmd += " ${args.dockerfile}"
+    return cmd
+}
+
+def imageBuild(Map args) {
+    sh buildCommand(args)
+}
+
+def imagePush(String image_name, String tag) {
+    sh "docker push ${image_name}:${tag}"
+}
+
+def imageRemove(String image_name, String tag) {
+    sh "docker rmi ${image_name}:${tag}"
+}
+
+def imageTag(String image_name, String current_tag, String new_tag) {
+    sh "docker tag ${image_name}:${current_tag} ${image_name}:${new_tag}"
+}
+
+def imageTagPush(String image_name) {
+    hubLogin()
```

Comment:
Since we're wrapping the `docker login` command with a try/catch...wont we just continue on here and then fail on the next line? Should we just let it raise out so that we its more explicit that we failed to log in?

Reply:
To your point, it's just a matter of where a failure happens _if_ a failure happens. I like the idea of continuing on in the case that the daemon is already logged in but someone either didn't provide the environment variables or they're somehow wrong.

Reply:
I agree with Alec.  If something is misconfigured, the system should fail.  Otherwise misconfigurations could go unnoticed indefinitely.